lib/Sema/CSBindings.cpp

```diff
@@ -1567,18 +1567,27 @@ void PotentialBindings::retract(Constraint *constraint) {
   EquivalentTo.remove_if(hasMatchingSource);
 }
 
-LiteralBindingKind BindingSet::getLiteralKind() const {
-  LiteralBindingKind kind = LiteralBindingKind::None;
-
+SmallVector<KnownProtocolKind> BindingSet::getAllProtocolKinds() const {
+  SmallVector<KnownProtocolKind> kind;
   for (const auto &literal : Literals) {
     auto *protocol = literal.first;
     const auto &info = literal.second;
 
     // Only uncovered defaultable literal protocols participate.
     if (!info.viableAsBinding())
       continue;
+    
+    if (auto protocolKind = protocol->getKnownProtocolKind())
+      kind.push_back(protocolKind.getValue());
+  }
+  return kind;
+}
 
-    switch (*protocol->getKnownProtocolKind()) {
+LiteralBindingKind BindingSet::getLiteralKind() const {
```

Comment:
Let's rename this method into something like `getLiteralForScore()` and leave a comment in the CSBindings.h that this method returns a literal requirement that has most impact on the binding score.